Classes/Service/WebAuthnAuthenticationService.php

```diff
@@ -40,31 +46,64 @@ class WebAuthnAuthenticationService extends AuthenticationService
      */
     private $webAuthnSession;
 
+    /**
+     * @var array
+     */
+    private $backendExtensionConfiguration;
+
     public function __construct()
     {
         $this->backendUserRepository = GeneralUtility::makeInstance(ObjectManager::class)
             ->get(BackendUserRepository::class);
         $this->webAuthnService = WebAuthnServiceFactory::fromGlobals();
         $this->webAuthnSession = GeneralUtility::makeInstance(WebAuthnSession::class);
+        $this->backendExtensionConfiguration = GeneralUtility::makeInstance(ExtensionConfiguration::class)->get('cvc_webauthn');
+    }
+
+    public function processLoginData(array &$loginData, $passwordTransmissionStrategy)
+    {
+        $isProcessed = false;
+        if ($passwordTransmissionStrategy === 'normal') {
+            $loginData['webauthn-uident'] = GeneralUtility::_GP('webauthn-userident');
+            if ($loginData['uident'] == '') {
+                $loginData['uident'] = $loginData['webauthn-uident'];
+            }
+            $isProcessed = true;
+        }
+
+        return $isProcessed;
     }
 
     /**
      * @throws \TYPO3\CMS\Core\Crypto\PasswordHashing\InvalidPasswordHashException
      */
     public function authUser(array $user): int
     {
-        $username = (string) $user['username'];
-        $data = base64_decode($this->login['uident']);
+        if ($this->backendExtensionConfiguration['secondFactorLogin'] == '1') {
+            $this->serviceChainValue = parent::authUser($user);
```

Comment:
It looks like `$this->serviceChainValue` is only used as a local variable here and like it can be a simple local variable then.

```suggestion
            $serviceChainValue = parent::authUser($user);
```

Reply:
global variable changed to local